community/planning/griddle/openapi.yaml

```diff
@@ -0,0 +1,2146 @@
+            distributed ledger. The Grid Daemon processes these events and updates
+            the Grid database accordingly.
+            This means that there is a delay between when transactions are
+            submitted and when the new data is actually available to fetch using
+            the Grid REST API GET endpoints. This delay can range from a few
+            milliseconds to several minutes depending on the underlying ledger.
+            Developers should take this delay into account when designing apps
+            on top of Grid.
+          content:
+            application/json:
+              schema:
+                properties:
+                  link:
+                    $ref: "#/components/schemas/BatchPostResponse"
+        "400":
+          $ref: "#/components/responses/400BadRequest"
+        "500":
+          $ref: "#/components/responses/500ServerError"
+        "503":
+          $ref: "#/components/responses/503ServiceUnavailable"
+
+  /organizations/{org_id}/agents/{public_key}:
+    get:
+      tags:
+        - Pike
+      summary: Fetches an agent with the given public key
+      operationId: get_agent
+      parameters:
+        - name: org_id
+          in: path
+          description: ID of the organization to fetch
+          required: true
+          schema:
+            type: string
+        - name: public_key
+          in: path
+          description: Public key of the agent to fetch
+          required: true
+          schema:
+            type: string
+        - $ref: "#/components/parameters/service_id"
+      responses:
+        "200":
+          description: |
+            Successful request. The response will include a JSON object
+            representing the agent.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Agent"
+        "400":
+          $ref: "#/components/responses/400BadRequest"
+        "404":
+          $ref: "#/components/responses/404NotFound"
+        "500":
+          $ref: "#/components/responses/500ServerError"
+        "503":
+          $ref: "#/components/responses/503ServiceUnavailable"
+    put:
+      tags:
+        - Pike
+      summary: |
+        Updates a single agent with the given public key belonging to the
+        organization with the given org ID
+      operationId: put_agent
+      parameters:
+        - name: org_id
+          in: path
+          description: ID of the organization to the agent belongs to
+          required: true
+          schema:
+            type: string
+        - name: public_key
+          in: path
+          description: Public key of the agent to update
+          required: true
+          schema:
+            type: string
+        - $ref: "#/components/parameters/service_id"
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/PostAgents"
+        description: A JSON list of Agents
```

Comment:
Same question that I asked for the orgs endpoint here, across all post endpoints probably.